packages/gasket-helper-intl/lib/index.d.ts

```diff
@@ -29,6 +29,17 @@ export interface LocaleManifest {
  * or a URL template with a `:locale` path param to a .json file.
  */
 export type LocalePathPart = string;
+
+/**
+ * Callback which receives a context object for resolving a LocalePathPath
+ */
+export type LocalePathThunk = (context: {}) => LocalePathPart;
```

Comment:
Do we have better details on what's in the context object?

Reply:
It's fairly varied to try and type out completely. I added some notes